javascript/first-js-oop-project.md

```diff
@@ -0,0 +1,150 @@
+6. Now push the objects created into the array. In order to push objects in the array, do the following.
+   ```javascript
+      event_array.push(event_obj1);
+      event_array.push(event_obj2);
+      event_array.push(event_obj3);
+   ```
+7. Keep checking everytime you push the element using ***console.log***.
+   ```javascript
+       //inorder to check whether the elements are pushed, Use Console.log
+       console.log(event_array);
+   ```
+
+8. After that you have created the array now we need to write the jQuery code to iterate through it in the same js file event.js .
+   - The jQuery library is a single JavaScript file, and you reference it with the HTML.
+      ```html
+        <head>
+         <script src="jquery-3.4.0.min.js"></script>
+        </head>
+      ```
+- The following code is the syntax to iterate through an array.
+  ```javascript
+     $.each([ 52, 97 ], function( index, value ) {
+     alert( index + ": " + value );
+     });
+  ```
+- If an object is used as the collection, the callback is passed a key-value pair each time:
+  ```javascript
+     var obj = {
+     "KLOS Golden Gala": "An evening with hollywood vampires",
+     "Jenny Lewis": "On the line tour 2019"};
+    $.each( obj, function( key, value )
+     {
+      alert( key + ": " + value );
+     });
+  ```
+7. Below is the jQuery code to iterate through the event_array that you have already created.
+    - `.each()` is used to iterate through the array of objects. Above code represents the syntax to iterate using `.each()`
+    - `.html()` is used to return the HTML code from jQuery to the target element of the HTML page.
+    - **event** is the target element in the below code.
+      ```javascript
+       $(document).ready(function()
+        {
+          var html = "";
+          // Here index represents the position of each object and item here represents the object
+          $.each(event_array, function(index, item)
+           {
+             //Here subItemName and subItem represents the key value pair to iterate through each object in the array
+             $.each(item, function(subItemName, subItem)
+              {
+                html += ("<li>"+ subItem + "</li>");
+             });
+           });
+        // insert final html into #event...
+       $("#event").html(html);
+       });
+      ```
++ Things to notice:
+   - All jQuery methods in our examples, are inside a document ready event.
+   - This is to prevent any jQuery code from running before the document is finished loading(is ready).
+
+8. Create an HTML file named `index.html`. This is the HTML file containing the div tag or that can be any other element as per the requirement.
+   ```html
+      <!DOCTYPE html>
+      <html>
+       <head>
+         <title></title>
+         <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
+      </head>
+      <body>
+            <div id="events">
+                    <!-- EVENTS GET INSERTED HERE -->
```

Comment:
Great!  Now you will need an unordered list (ul) here for your li items

Reply:
@alodahl  I am already passing unordered list through .html(). Hence it is not required to change it here. We will need div or another element to display the unordered list. Even then if you feel it better to change the div element do mention. I shall change it then.

Reply:
Thank you for this fix. We need to set a good example for beginners that li tags should be in ol or ul. There’s nothing wrong with a div , but we needed ul tags too.